org.mosh.mosh.json

```
@@ -0,0 +1,129 @@
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "18.08",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "mosh",
+  "separate-locales": false,
+  "finish-args": [
+    "--share=network",
+    "--talk-name=org.freedesktop.Flatpak"
+  ],
+  "cleanup": [
+    "*.a",
+    "*.la",
+    "/include",
+    "/lib/pkgconfig",
+    "/share/man"
+  ],
+  "modules": [
+    {
+      "name": "protobuf",
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://github.com/google/protobuf/releases/download/v3.8.0/protobuf-all-3.8.0.tar.gz",
+          "sha256": "b7220b41481011305bf9100847cf294393973e869973a9661046601959b2960b"
+        }
+      ],
+      "cleanup": [
+        "/bin/protoc"
+      ]
+    },
+    {
+      "name": "perl",
+      "no-autogen": true,
+      "config-opts": [
+        "-des"
+      ],
+      "cleanup": [
+        "/bin/perl5*",
+        "/bin/c2ph",
+        "/bin/corelist",
+        "/bin/cpan",
+        "/bin/enc2xs",
+        "/bin/encguess",
+        "/bin/h2*",
+        "/bin/instmodsh",
+        "/bin/json_pp",
+        "/bin/libnetcfg",
+        "/bin/perlbug",
+        "/bin/perldoc",
+        "/bin/perlivp",
+        "/bin/perlthanks",
+        "/bin/piconv",
+        "/bin/pl2pm",
+        "/bin/pod*",
+        "/bin/prove",
+        "/bin/pstruct",
+        "/bin/ptar*",
+        "/bin/shasum",
+        "/bin/splain",
+        "/bin/xsubpp",
+        "/bin/zipdetails",
+        "/lib/perl5/site_perl",
+        "/man",
+        "*.h",
+        "*.pod"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://www.cpan.org/src/5.0/perl-5.28.0.tar.gz",
+          "sha256": "7e929f64d4cb0e9d1159d4a59fc89394e27fa1f7004d0836ca0d514685406ea8"
+        },
+        {
+          "type": "script",
+          "dest-filename": "configure",
+          "commands": [
+            "exec ./configure.gnu $@"
+          ]
+        }
+      ],
+      "post-install": [
+        "find /app/lib/perl5 -type f -exec chmod u+w {} \\;"
+      ]
+    },
+    {
+      "name": "mosh",
+      "config-opts": [
+        "--disable-server"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://mosh.org/mosh-1.3.2.tar.gz",
+          "sha256": "da600573dfa827d88ce114e0fed30210689381bbdcff543c931e4d6a2e851216"
+        },
+        {
+          "type": "patch",
+          "path": "fake-proxy-flatpak.patch"
+        },
+        {
+          "type": "file",
+          "path": "org.mosh.mosh.appdata.xml"
+        }
+      ],
+      "post-install": [
+        "install -Dm644 org.mosh.mosh.appdata.xml /app/share/appdata/org.mosh.mosh.appdata.xml"
+      ]
+    },
+    {
+      "name": "ssh-wrapper",
+      "buildsystem": "simple",
+      "sources": [
+        {
+          "type": "script",
+          "dest-filename": "ssh",
+          "commands": [
+            "exec flatpak-spawn --host --forward-fd=1 --forward-fd=2 --forward-fd=3 ssh \"$@\""
```

```c
  for (i = 0; forward_fds != NULL && forward_fds[i] != NULL; i++)
    {
      int fd = strtol (forward_fds[i],  NULL, 10);
      gint handle = -1;

      if (fd == 0)
        {
          g_printerr ("Invalid fd '%s'\n", forward_fds[i]);
          return 1;
        }

      if (fd >= 0 && fd <= 2)
        continue; // We always forward these
```

so you could do

```suggestion
            "exec flatpak-spawn --host ssh --forward-fd=3 \"$@\""
```

but I think it's fine to be explicit.

---

Some thinko there anyway, it should be FDs 0-2, not 1-3. So we can remove all of those --forward-fd args.